pallets/circuit/src/lib.rs

```diff
@@ -1441,7 +1441,11 @@ impl<T: Config> Pallet<T> {
 
     fn kill(local_ctx: &mut LocalXtxCtx<T>, cause: CircuitStatus) {
         local_ctx.xtx.status = cause.clone();
-        Optimistic::<T>::try_slash(local_ctx);
+        if let CircuitStatus::RevertTimedOut = cause {
```

Comment:
prefer match cause for CircuitStatus::DroppedAtBidding =>  Optimistic::<T>::try_dropped_at_bidding_refund(local_ctx); and Optimistic::<T>::try_slash(local_ctx) for everything else

Reply:
very true